apps/blockchain/lib/blockchain/ethash/rand_memo_hash.ex

```diff
@@ -0,0 +1,58 @@
+defmodule Blockchain.Ethash.RandMemoHash do
+  alias Blockchain.Ethash
+  alias ExthCrypto.Hash.Keccak
+
+  use Bitwise
+
+  @doc """
```

Comment:
Awesome, will review more thoroughly but great work getting this done. We should make a note that none of our crypto algorithms are “safe” in the sense of resistance to timing or cache attacks, but since we’re not mining we should be okay with correct but not “safe.”